engine/src/main/java/org/pentaho/di/trans/steps/textfileoutput/TextFileOutput.java

```diff
@@ -1034,6 +1019,28 @@ private void createParentFolder( String filename ) throws Exception {
     }
   }
 
+  boolean isWriteEnclosureForFieldName( ValueMetaInterface v, String fieldName ) {
+    return ( isWriteEnclosed( v ) )
+            || ( !meta.isEnclosureFixDisabled() && containsSeparatorOrEnclosure( fieldName.getBytes(),
```

Comment:
DRY (Do Not Repeat Yourself)
this code snippet below is repeated:
`( !meta.isEnclosureFixDisabled() && containsSeparatorOrEnclosure( fieldName.getBytes(),
            data.binarySeparator, data.binaryEnclosure ) );`
to avoid same the issue that originated this jira issue where one logic for for isWriteEnclosure get's updated and the others don't. Create a wrapper function similar to how` isWriteEnclosed( ValueMetaInterface v )`